p2p/src/network.rs

```diff
@@ -167,13 +166,12 @@ impl_handle_methods! {
 /// Base network layer structure, holding connections interacting with peers.
 struct NetworkBase<T: Pload, K: Kex, E: Enc> {
     /// Listening address for incoming connections. Must parse into [`std::net::SocketAddr`]
-    listen_addr: String,
+    listen_addr: SocketAddr,
     /// Current [`Peer`]s in [`Peer::Ready`] state.
     peers: HashMap<PublicKey, RefPeer<T>>,
     /// Map from [`std::net::IpAddr`] of the untrusted remote [`Peer`]:
     /// inserted by [`DisconnectPeer`] and removed by [`ConnectPeer`] from Sumeragi.
-    /// In case the [`String`] represents an unresolved hostname, the first reconnection is not refused
-    untrusted_peers: HashSet<String>,
+    untrusted_peers: HashSet<IpAddr>,
```

Comment:
Looks like we need to think about equality among `IpAddr` instances. For example should the peer `127.0.0.1` be equal to `localhost` equal to `::1/128`?

Reply:
Similarly, the loopback is different to `localhost` but only in very specific circumstances. We need to be able to resolve the addresses somehow.

Reply:
We're not dealing with hostnames here, so that one is not a problem to consider.

Comparing supposedly equal IPs sounds quite complicated. One could run two peers listening on `127.0.0.1` and `::1` respectively for example. I'd say we should just leave it as is.